iceoryx_posh/source/roudi/roudi_config_json_file_provider.cpp

```diff
@@ -0,0 +1,181 @@
+            }
+        }
+        else
+        {
+            m_customConfigFilePath = cmdLineArgs.configFilePath;
+        }
+    }
+}
+
+iox::cxx::expected<iox::mepoo::MePooConfig, iox::roudi::RouDiConfigFileParseError>
+JsonRouDiConfigFileProvider::getMempool(json_t const* segment)
+{
+    iox::mepoo::MePooConfig mempoolConfig;
+    json_t const* mempools = json_getProperty(segment, "mempool");
+    if (!(mempools && JSON_ARRAY == json_getType(mempools)))
+    {
+        return iox::cxx::error<iox::roudi::RouDiConfigFileParseError>(
+            iox::roudi::RouDiConfigFileParseError::SEGMENT_WITHOUT_MEMPOOL);
+    }
+    json_t const* mempool;
+    uint32_t poolCount = 0;
+    for (mempool = json_getChild(mempools); mempool != 0; mempool = json_getSibling(mempool))
+    {
+        if (JSON_OBJ == json_getType(mempool) && poolCount < iox::MAX_NUMBER_OF_MEMPOOLS)
+        {
+            json_t const* chunkSize = json_getProperty(mempool, "size");
+            if (!(chunkSize && JSON_INTEGER == json_getType(chunkSize)))
+            {
+                return iox::cxx::error<iox::roudi::RouDiConfigFileParseError>(
+                    iox::roudi::RouDiConfigFileParseError::MEMPOOL_WITHOUT_CHUNK_SIZE);
+            }
+            json_t const* chunkCount = json_getProperty(mempool, "count");
+            if (!(chunkCount && JSON_INTEGER == json_getType(chunkCount)))
+            {
+                return iox::cxx::error<iox::roudi::RouDiConfigFileParseError>(
+                    iox::roudi::RouDiConfigFileParseError::MEMPOOL_WITHOUT_CHUNK_COUNT);
+            }
+            uint32_t size = static_cast<uint32_t>(json_getInteger(chunkSize));
+            uint32_t count = static_cast<uint32_t>(json_getInteger(chunkCount));
+            mempoolConfig.addMemPool({size, count});
+        }
+        poolCount++;
+    }
+    if (poolCount > iox::MAX_NUMBER_OF_MEMPOOLS)
+    {
+        return iox::cxx::error<iox::roudi::RouDiConfigFileParseError>(
+            iox::roudi::RouDiConfigFileParseError::MAX_NUMBER_OF_MEMPOOLS_PER_SEGMENT_EXCEEDED);
+    }
+    return iox::cxx::success<iox::mepoo::MePooConfig>(mempoolConfig);
+}
+
+iox::cxx::expected<iox::RouDiConfig_t, iox::roudi::RouDiConfigFileParseError> JsonRouDiConfigFileProvider::parse()
+{
+    json_t mem[NUMBER_OF_JSON_NODES];
+    std::string content, line;
+
+    // Early exit in case no config file path was provided
+    if (m_customConfigFilePath.empty())
+    {
+        iox::RouDiConfig_t defaultConfig;
+        defaultConfig.setDefaults();
+        return iox::cxx::success<iox::RouDiConfig_t>(defaultConfig);
+    }
+
+    iox::cxx::FileReader reader(m_customConfigFilePath, "", iox::cxx::FileReader::ErrorMode::Inform);
+    while (reader.ReadLine(line))
+    {
+        content.append(line);
+    }
+    json_t const* json = json_create(&content[0], mem, NUMBER_OF_JSON_NODES);
+    auto groupOfCurrentProcess = iox::posix::PosixGroup::getGroupOfCurrentProcess().getName();
```

Comment:
Please place this line right before the for-loop it is used in. Improves readability.